File: Leetcode/1189.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        constexpr int MAXN = 26;
        int stats[MAXN]{};
        for (const auto &i : text) ++stats[i-'a'];
        int counter = 0;
        while (true) {
            if (stats['b'-'a'] > 0) {
                --stats['b'-'a'];
            } else {
                break;
            }
            if (stats['a'-'a'] > 0) {
                --stats['a'-'a'];
            } else {
                break;
            }
            if (stats['l'-'a'] > 1) {
                stats['l'-'a'] -= 2;
            } else {
                break;
            }
            if (stats['o'-'a'] > 1) {
                stats['o'-'a'] -= 2;
            } else {
                break;
            }
            if (stats['n'-'a'] > 0) {
                --stats['n'-'a'];
            } else {
                break;
            }
            ++counter;
        }
        return counter;
    }
};

File: Leetcode/1189/1189.go
func min(a ... int) int {
	ans := math.MaxInt32
	for _, i := range a {
		if i < ans {
			ans = i
		}
	}
	return ans
}

func maxNumberOfBalloons(text string) int {
	const MAXN = 26
	var freq [MAXN]int
	for i := 0; i < len(text); i++ {
		freq[text[i]-'a']++
	}
	minVal := math.MaxInt32
	const STR = "balloon"
	if freq['b'-'a'] < 1 || freq['a'-'a'] < 1 || freq['l'-'a'] < 2 || freq['o'-'a'] < 2 || freq['n'-'a'] < 1 {
		return 0
	}
	minVal = min(minVal, freq['b'-'a'], freq['a'-'a'], freq['l'-'a']>>1, freq['o'-'a']>>1, freq['n'-'a'])
	return minVal
}